src/addon/ESMX/Comps/ESMX_Data/ESMX_Data.F90

```diff
@@ -148,6 +153,45 @@ subroutine SetServices(xdata, rc)
       localPet=xstate%myid, rc=rc)
     if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
       line=__LINE__, file=__FILE__)) return
+
+    ! validate config
+    call ESMF_GridCompGet(xdata, name=compLabel, configIsPresent=isFlag, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+      line=__LINE__, file=__FILE__)) return  ! bail out
+    if (isFlag) then
+      ! Config present, assert it is in the ESMX YAML format
+      call ESMF_GridCompGet(xdata, config=config, rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+        line=__LINE__,    file=__FILE__)) return  ! bail out
+      call ESMF_ConfigGet(config, hconfig=hconfig, rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+        line=__LINE__, file=__FILE__)) return  ! bail out
+      hconfigNode = ESMF_HConfigCreateAt(hconfig, keyString=compLabel, rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+        line=__LINE__, file=__FILE__)) return  ! bail out
+      ! component responsibility to validate ESMX handled options here, and
+      ! potentially locally handled options
+      isFlag = ESMF_HConfigValidateMapKeys(hconfigNode, &
```

Comment:
I'm okay with this but it will have to evolve as the component configuration keys for the driver evolves, that could be tricky. It also ties ESMX Data to ESMX Driver more than I had wanted.

Reply:
Yes, I was thinking about this, too. In the end I did not see any other way than to say, it is the components responsibility to check the validity of all of the keys specified under *its* section. Only the component itself knows the full list of valid keys. Of course a component can be sloppy and not validate the keys... it will work, but might not catch user errors like typos.